src/Tokenizer.js

```diff
@@ -457,17 +457,33 @@ module.exports = class Tokenizer {
   link(src) {
     const cap = this.rules.inline.link.exec(src);
     if (cap) {
-      const lastParenIndex = findClosingBracket(cap[2], '()');
-      if (lastParenIndex > -1) {
-        const start = cap[0].indexOf('!') === 0 ? 5 : 4;
-        const linkLen = start + cap[1].length + lastParenIndex;
-        cap[2] = cap[2].substring(0, lastParenIndex);
-        cap[0] = cap[0].substring(0, linkLen).trim();
-        cap[3] = '';
+      const trimmedUrl = cap[2].trim();
+      if (!this.options.pedantic && trimmedUrl.startsWith('<')) {
+        // commonmark requires matching angle brackets
+        if (!trimmedUrl.endsWith('>')) {
+          return;
+        }
+
+        // ending angle bracket cannot be escaped
+        const rtrimSlash = rtrim(trimmedUrl.slice(0, -1), '\\');
+        if ((trimmedUrl.length - rtrimSlash.length) % 2 === 0) {
+          return;
+        }
```

Comment:
That regex wouldn't catch an escaped slash before the escaped bracket (e.g. `\\\>`) and the regex to catch that is vulnerable to ReDOS. That is why we have the rtrim function instead of using replace with a regex.

Reply:
Wouldn't the `[^\]` handle the case of the escaped slash? If there is an extra `\` then it won't match.

For my own understanding, can you help me with the Regex vulnerability of that pattern? `[^a]aa$`

Reply:
reversing the string would be much slower on big strings. I know this isn't as pretty as regex replace but it is the fastest way I could think of while still being simple.

Reply:
If this seems like the fastest way, I'm ok with it. Just wanted to be sure.